### ISSUE TITLE
Timezone issues with new calendar version

### ISSUE THREAD

**Issue by kamuffe** [stefanuebe/vaadin_fullcalendar#136]
Hi Stefan.

I migrated to the new reworked calendar and have some issues with the timezones.

My Setup is the following: 
**JVM TimeZone:**
ZoneId.systemDefault: Europe/Berlin

**FullCalendar TimeZones:**
fullCalendar.setTimezone( Timezone.getSystem() );

Output:
Server Side Calendar Timezone:
Timezone{clientSideValue='Europe/Berlin', zoneId=Europe/Berlin}

Timezone retrieved from Client (via fullCalendar.addBrowserTimezoneObtainedListener):
Timezone{clientSideValue='Europe/Berlin', zoneId=Europe/Berlin}

Now the issue:

When I select a time range in the front end calendar, wrong times are given to the server.
After I select time 15:55 - 16:50 (Client Berlin time) 

This request goe to the server:

**POST Request**
```
{
	"clientId": 13,
	"csrfToken": "aToken",
	"rpc": [
		{
			"data": {
				"event.detail.allDay": false,
				"event.detail.end": "2022-02-10T16:50:00Z",
				"event.detail.start": "2022-02-10T15:55:00Z"
			},
			"event": "select",
			"node": 407,
			"type": "event"
		}
	],
	"syncId": 16
}
```
Why is the request interpreted as UTC?

FrontEnd calendar configuration:
```
...
    _createInitOptions(initialOptions) {

        let options = {
	        // Sets the height of the view area of the calendar.
            contentHeight: '100%',
            // local browser timezone, needed to show nowIndicator in correct time
			timeZone: 'local',
            initialView: 'timeGridWeek', // added to avoid two calls: first get all events for month, seconds get all events for week
            allDayContent: { html: '' },
            nowIndicator: true,
            // If the rows of a given view don’t take up the entire height, they will expand to fit.
            expandRows: true,
			slotLabelFormat: {
				hour: '2-digit',
				minute: '2-digit',
				omitZeroMinute: false
			},
            // // no native control elements
            headerToolbar: false,
            weekNumbers: true,
            eventLimit: this.eventLimit,
            navLinks: this.navLinks,
            selectable: this.selectable,
        };

        return super._createInitOptions(options);
    }
...
```

Did I miss something?

Thanks a lot for your work.

Best,
Kamuffe

**Comment by kamuffe** [comment]
Hi @stefanuebe 

I was able to fix the wrong POST request by adopting the init option on the extended client side full-calendar.
There was simply a missing client side migration. Your wiki helped here ;)

But on server side now I'm a little confused.

**Steps:**
User selects start 2022-02-10T16:55:00 Berlin
Client sends converted UTC start = 2022-02-10T15:55:00Z

![FC-Timezone](https://user-images.githubusercontent.com/5285367/153606635-1db6b323-31d4-466f-b90d-ce5774d1164e.PNG)

So far so good.

Server retrieves data in a TimeslotsSelectedEvent (calendar.addTimeslotsSelectedListener(TimeslotsSelectedEvent event)):

```
System.out.println( "ZoneId: " + event.getSource().getTimezone().getZoneId() );
ZoneId: Europe/Berlin

LocalDateTime startDate = event.getStart();

// correct, interpreted as UTC as it is
System.out.println( "getStart(): " + event.getStart() );
getStart(): 2022-02-10T15:55

// wrong, UTC is 2022-02-10T15:55:00Z
System.out.println( "getStartWithOffset():  " + event.getStartWithOffset() );
getStartWithOffset(): 2022-02-10T14:55:00Z

// convert from org.vaadin.stefan.fullcalendar.Timezone.convertToInstant(LocalDateTime)
// startDate.toInstant( event.getSource().getTimezone().getZoneId().getRules().getOffset( startDate ) )
// wrong UTC is 2022-02-10T15:55:00Z
FC convert: 2022-02-10T14:55:00Z

// desired result, where ZoneId.systemDefault() is Berlin TimeZone
System.out.println( "custom: "+ startDate.atZone( ZoneId.of( "UTC" ) ).withZoneSameInstant( ZoneId.systemDefault() ).toLocalDateTime() );
custom: 2022-02-10T16:55
```

Is there still a wrong conversion in org.vaadin.stefan.fullcalendar.Timezone.convertToInstant(LocalDateTime) or do I miss something?

Thanks a lot.
Best,
Kamuffe

**Comment by stefanuebe** [comment]
Hmmm, it simply might be, that I simply oversaw those methods and forgot to migrate that event to the new api. From simply seeing the method I would expect a LocalDateTime with the Berlin offset applied (+1h), correct.

**Comment by stefanuebe** [comment]
Going to fix it and release a new version in a minute.

**Comment by stefanuebe** [comment]
Fixed with Version [4.1.1](https://vaadin.com/directory/component/full-calendar-flow). 

Maybe that's a good example for that the events should also be covered with junit tests. Anyone here who likes writing automated junit tests? 😄

**Comment by kamuffe** [comment]
@stefanuebe 
I tested the new version and it fixed my issue.

Thanks a lot.

I try to contribute some tests this week :)

Best,
Kamuffe
